src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -443,6 +448,15 @@ public override void AttachToDependencyGraph(DependencyAnalyzerBase<NodeFactory>
             RuntimeFunctionsGCInfo = new RuntimeFunctionsGCInfoNode();
             graph.AddRoot(RuntimeFunctionsGCInfo, "GC info is always generated");
 
+            if (_ibcTuning)
+            {
+                ProfileDataSection = new ProfileDataSectionNode();
+                // Don't add to root and R2R Header here, as if we don't actually compile any methods, we cannot have this section
```

Comment:
I don't think the logic associated with this and the comment block is necessary.

`ArrayOfEmbeddedDataNode` overrides `ShouldSkipEmittingObjectNode` to return true if there's no entries. The ready to run header node skips adding references to sections that report `ShouldSkipEmittingObjectNode` as true.

I think we can just add to the header here, remove the OnMarked logic that adds the node to header, and remove the lock added in Header.Add.

Reply:
Ah, I missed that part of the R2R header node processing. I'd say that's a bit simpler  :)